### PR TITLE
Add Java Record support in JavaObjectTransformer

### PR DESCRIPTION
**Pull request by panilya**
At the moment, Datafaker doesn't support Java Records. This PR addresses this problem

### CODE REVIEW

File: src/main/java/net/datafaker/transformations/JavaObjectTransformer.java [hunk 1]
```diff
@@ -23,7 +25,32 @@ public Object apply(Object input, Schema<Object, ?> schema) {
...
+                    .toArray(Class<?>[]::new);
+
+                final Field<Object, ?>[] fields = schema.getFields();
+                final Object[] values = new Object[fields.length];
+                for (int i = 0; i < fields.length; i++) {
+                    values[i] = fields[i].transform(result);
+                }
+
+                try {
+                    result = clazz.getDeclaredConstructor(componentTypes).newInstance(values);
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/transformations/JavaObjectTransformer.java:46]: it's better to cache it as well to avoid constant calculation logic

Comment (panilya) [reply src/main/java/net/datafaker/transformations/JavaObjectTransformer.java:46]: I don't it's possible to cache object, but I can cache constructor. WDYT?

Comment (snuyanzin) [reply src/main/java/net/datafaker/transformations/JavaObjectTransformer.java:46]: yes, constructor should be enough

Comment (panilya) [reply src/main/java/net/datafaker/transformations/JavaObjectTransformer.java]: @snuyanzin I have a question for you, imagine an article about Datafaker 2.0, what would you expect to see in that article, I mean topics or something like this

**Review by snuyanzin — APPROVED** [review]

File: src/main/java/net/datafaker/transformations/JavaObjectTransformer.java (hunk 1 above)
Comment (snuyanzin) [reply src/main/java/net/datafaker/transformations/JavaObjectTransformer.java]: depends on the target audience...

there is a list of topics missed in faker related talks

1. custom providers 
2. there are ways to generate sequences (stream and sequence) 
3. the transformation approach (csv, json, sql, java, custom)
4. object population as a replacement for builders

Comment (panilya) [reply src/main/java/net/datafaker/transformations/JavaObjectTransformer.java]: Could I send you my draft of the article about Datafaker 2.0? I would like to hear your opinion if you don't mind and you have time, of course

Comment (snuyanzin) [reply src/main/java/net/datafaker/transformations/JavaObjectTransformer.java]: yep, sure

**Comment by bodiam** [comment]
Nice feature Illia, amazing work!

**Pull request merged**